scala-thoughts.md

```diff
@@ -0,0 +1,5 @@
+I really like scala so far. So far it seems like it gives me everything I liked about functional Python and Haskell with the benefits of a static type system over Python and flexibility over Haskell. Parts of it remind me of Java and I feel like they're mostly vestiges from it being made in Java, and they don't really affect the feel fo the language very much. I actually don't really have any qualms at all right now; it seems like a good language. 
+
+I have plenty of things that I'm not sure about. What's the difference between a seq, and a list, and a type with an asterisk at the end? They all seem to function the same, i.e. like a Python list, at least for the first two. I don't feel like I ever treated seqs and lists differently throughout this assignment even though they were both used a lot. I'm still not sure about style conventions within Scala and I'm not sure if I've been following them or horribly mangling them. I'm not sure to what degree I should be writing one-liners (which I ~~probably~~ definitely did too much but I like writing one-liners). Comparing mine to the lab's solutions further confuses this because they did plenty of things that I think could have been done better (like checkValuesIncrease in recursionpatternmatchingexercise) or that I have no idea how they work (like compress, which looks like it would turn (1,1,2,3,1,1) into (1,2,3,1) rather than (1,2,3)). This also goes for Ben's solution. That said looking at both solutions has given me a better understanding of some of the built-in functions like span and partition and when they might be useful.
+
```

Comment:
I completely agree about the different list types. I looked into them a little bit but mostly used them interchangeably. In terms of things with asterisks at the end, treating them like lists seemed to work within the context of this assignment but I doubt that's the correct approach to dealing with them in any case. I don't think you did one-liners too much, that's part of the fun of assignments like this. It's also a good challenge for learning the list comprehensions even better and it seems like you got a good grasp on them throughout the assignment.
In general, your solutions seemed very Scala-y (to use Prof Ben's wording) since none of us has an incredibly firm grasp on the language or all of its capabilities. I think there were a few places you could have been more elegant or efficient, but I'm sure the same can be said for any of our code. Part of that is personal preference, meaning that I can find a few pieces of code in the sample solutions that I also disagree with and would like to see implemented differently. The great part about this assignment is that we had a lot of flexibility on how we solved each of the methods and that we got to learn about the different options available to us in Scala.